packages/components/src/systems/PatientMedHistory/index.tsx

```diff
@@ -62,17 +82,34 @@ export default function PatientMedHistory(props: PatientMedHistoryProps) {
 
   const baseURL = createMemo(() => `${client?.clinicalUrl}/prescriptions/`);
 
-  const fetchPatient = async () => {
-    const { data } = await client!.apollo.query({
-      query: GET_PATIENT_MED_HISTORY,
-      variables: { id: props.patientId }
-    });
-    if (data?.patient?.medicationHistory) {
-      setMedHistory(data.patient.medicationHistory);
-    } else {
+  const queryOptions = createMemo(() => ({
+    variables: { id: props.patientId },
+    client: client!.apollo
+  }));
+
+  const patientMedHistory = createQuery<GetPatientResponse, { id: string }>(
+    GET_PATIENT_MED_HISTORY,
+    queryOptions
+  );
+
+  createEffect(() => {
+    const medicationHistory = patientMedHistory()?.patient?.medicationHistory;
+    if (medicationHistory) {
+      const sortedMedHistory = medicationHistory.slice().sort((a, b) => {
+        const dateA = a?.prescription?.writtenAt
+          ? new Date(a.prescription.writtenAt).getTime()
+          : -Infinity;
+        const dateB = b?.prescription?.writtenAt
+          ? new Date(b.prescription.writtenAt).getTime()
+          : -Infinity;
+        return dateA - dateB;
+      });
```

Comment:
Are these date strings or dates already? I think you can just do `.sort()`?

Reply:
It's a date string at the moment or possibly undefined